### PR TITLE
Switch OTC Fork to OTel Distro for Traces

### PR DESCRIPTION
**Pull request by mat-rumian**
##### Description

PR is switching Tracing collection from OTC Fort to OTel Distro

---

##### Checklist

- [ ] Changelog updated

###### Testing performed

- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,10 +7,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ## [Unreleased]
 
+This release changes the OpenTelemetry Collector binary used for traces collection ([#2334]).
+This change affects you only if you have enabled traces with `sumologic.traces.enabled: true`
+AND you have customized the configuration in the `otelcol.config.processors.source` property.
+If you have modified these properties, make sure to compare the [new configuration][source_processor_new_config]
+with the [old configuration][source_processor_old_config] and apply corresponding changes to your config.
+
+[source_processor_old_config]: https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/v2.9.0/deploy/helm/sumologic/values.yaml#L3476-L3492
+[source_processor_new_config]: https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/deploy/helm/sumologic/values.yaml#L3507-L3522
```
Comment (sumo-drosiek) [inline CHANGELOG.md:17]: I would rather go with specific commit here instead of `m`in`

File: deploy/helm/sumologic/values.yaml [hunk 2]
```diff
@@ -3366,16 +3370,39 @@ otelagent:
...
         metrics:
-          receivers: [otlp]
+          receivers: [otlp, otlp/deprecated]
           processors: [memory_limiter, k8s_tagger, batch]
           exporters: [otlp]
 
 ## Configure otelcol
 otelcol:
+  ## Set the _sourceName metadata field in Sumo Logic.
+  source_metadata:
```
Comment (astencel-sumo) [inline deploy/helm/sumologic/values.yaml:3384]:
```suggestion
  source_metadata:
    ## Set the _sourceName metadata field in Sumo Logic.
```

Comment (astencel-sumo) [inline deploy/helm/sumologic/values.yaml:3384]: Is there a reason why the `source_metadata` property uses snake_case while the other properties use camelCase?

File: CHANGELOG.md (hunk 1 above)
Comment (astencel-sumo) [reply CHANGELOG.md:17]: I agree we should pin the `new config` link to a specific commit for now. When releasing a new version, we can change the commit to a tag.

File: deploy/helm/sumologic/values.yaml [hunk 3]
```diff
@@ -3366,16 +3370,39 @@ otelagent:
...
+    ## A regular expression for hosts.
+    ## Matching hosts will be excluded from Sumo. The logs will still be sent to FluentD.
+    excludeHostRegex: ""
+    ## A regular expression for namespaces.
+    ## Matching namespaces will be excluded from Sumo. The logs will still be sent to FluentD.
+    excludeNamespaceRegex: ""
+    ## A regular expression for pods.
+    ## Matching pods will be excluded from Sumo. The logs will still be sent to FluentD.
+    excludePodRegex: ""
+
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:3405]: Why do we multiplicate those values from fluentd?
In 3.0 we would like to move them to common place and multiplication now won't help with that

Comment (mat-rumian) [reply deploy/helm/sumologic/values.yaml:3405]: Becuase fluentd is basing on different values which are not working for OTel Distro source processor.: e.g. `%{namespace}` vs `%{k8s.namespace.name}`. I know that in 3.0 we have to have single place where metadata is configured.

File: deploy/helm/sumologic/values.yaml (hunk 2 above)
Comment (mat-rumian) [reply deploy/helm/sumologic/values.yaml:3384]: nah, there's no :) will change it 👍

File: deploy/helm/sumologic/values.yaml [hunk 4]
```diff
@@ -3282,8 +3282,8 @@ otelagent:
     ## Add custom annotations only to otelagent daemonset.
     podAnnotations: {}
     image:
-      repository: "public.ecr.aws/sumologic/opentelemetry-collector"
-      tag: "0.38.1-sumo"
+      repository: "public.ecr.aws/sumologic/sumologic-otel-collector"
+      tag: "0.51.0-sumo-0"
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:3286]:
```suggestion
      tag: "0.52.0-sumo-0"
```
:upside_down_face:

File: deploy/helm/sumologic/values.yaml [hunk 5]
```diff
@@ -3395,8 +3422,8 @@ otelcol:
     ## Add custom annotations only to otelcol deployment.
     podAnnotations: {}
     image:
-      repository: "public.ecr.aws/sumologic/opentelemetry-collector"
-      tag: "0.38.1-sumo"
+      repository: "public.ecr.aws/sumologic/sumologic-otel-collector"
+      tag: "0.51.0-sumo-0"
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:3426]:
```suggestion
      tag: "0.52.0-sumo-0"
```

File: deploy/helm/sumologic/values.yaml [hunk 6]
```diff
@@ -4529,8 +4554,8 @@ otelgateway:
     ## Add custom annotations only to otelagent daemonset.
     podAnnotations: {}
     image:
-      repository: public.ecr.aws/sumologic/sumologic-otel-collector
-      tag: 0.50.0-sumo-0
+      repository: "public.ecr.aws/sumologic/sumologic-otel-collector"
+      tag: "0.51.0-sumo-0"
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:4558]:
```suggestion
      tag: "0.52.0-sumo-0"
```

**Review by sumo-drosiek — APPROVED** [review]

**Review by astencel-sumo — APPROVED** [review]

**Pull request merged**